domain/model/state/controllerstate.go

```diff
@@ -1096,106 +1168,211 @@ AND       u.removed = false
+// by the uuid. The information returned is intended to augment the information
+// that lives in the model.
+// The following error types can be expected:
+// - [modelerrors.NotFound] when the model is not found for the given model
+// uuid.
+func (s *State) GetModelSummary(
+	ctx context.Context,
+	modelUUID coremodel.UUID,
+) (model.ModelSummary, error) {
 	db, err := s.DB()
 	if err != nil {
-		return nil, errors.Capture(err)
+		return model.ModelSummary{}, errors.Capture(err)
 	}
 
 	q := `
-SELECT    (p.access_type, m.uuid, m.name, m.cloud_name, m.cloud_region_name,
-          m.model_type, m.cloud_type, m.owner_name, m.cloud_credential_name,
-          m.cloud_credential_cloud_name, m.cloud_credential_owner_name,
-          m.life, mll.time, m.is_controller_model, m.controller_uuid) AS (&dbModelSummary.*)
-FROM      v_user_auth u
-JOIN      v_permission p ON p.grant_to = u.uuid
-JOIN      v_model m ON m.uuid = p.grant_on
-LEFT JOIN model_last_login mll ON m.uuid = mll.model_uuid AND mll.user_uuid = u.uuid
-WHERE     u.removed = false
-AND       u.name = $dbUserName.name
+SELECT (m.owner_name, ms.destroying, ms.cloud_credential_invalid,
+        ms.cloud_credential_invalid_reason, ms.migrating) AS (&dbModelSummary.*)
+FROM v_model_state ms
+JOIN v_model m ON m.uuid = ms.uuid
+WHERE ms.uuid = $dbModelUUID.uuid
 `
-	name := dbUserName{Name: userName.Name()}
-	modelStmt, err := s.Prepare(q, name, dbModelSummary{})
+	modelUUIDVal := dbModelUUID{UUID: modelUUID.String()}
+	stmt, err := s.Prepare(q, dbModelSummary{}, modelUUIDVal)
 	if err != nil {
-		return nil, errors.Capture(err)
+		return model.ModelSummary{}, errors.Capture(err)
 	}
 
-	var models []dbModelSummary
+	var modelSummaryVals dbModelSummary
 	err = db.Txn(ctx, func(ctx context.Context, tx *sqlair.TX) error {
-		err = tx.Query(ctx, modelStmt, name).GetAll(&models)
-		if errors.Is(err, sqlair.ErrNoRows) {
-			return nil
-		} else if err != nil {
-			return errors.Capture(err)
+		modelExists, err := s.checkModelExists(ctx, tx, modelUUID)
+		if err != nil {
+			return errors.Errorf(
+				"checking if model %q exists: %w", modelUUID.String(), err,
+			)
+		}
+		if !modelExists {
+			return errors.Errorf(
+				"model %q does not exist", modelUUID.String(),
+			).Add(modelerrors.NotFound)
+		}
+
+		err = tx.Query(ctx, stmt, modelUUIDVal).Get(&modelSummaryVals)
+		if err != nil {
+			return errors.Errorf(
+				"getting model %q summary: %w", modelUUID.String(), err,
+			)
 		}
 		return nil
 	})
+
 	if err != nil {
-		return nil, errors.Errorf("getting model summaries for user: %w", err)
+		return model.ModelSummary{}, errors.Capture(err)
 	}
 
-	modelSummaries := make([]coremodel.UserModelSummary, len(models))
-	for i, m := range models {
-		modelSummaries[i], err = m.decodeUserModelSummary()
-		if err != nil {
-			return nil, errors.Errorf("getting model summaries for user: %w", err)
-		}
+	ownerName, err := user.NewName(modelSummaryVals.OwnerName)
```

Comment:
I wish we had a version of this that didn't return an error - what's in the db is validated on the way in

Reply:
Would be nice. Let's discuss it this week.